src/app/modules/datagrid/datagrid-controller.js

```diff
@@ -1,135 +1,252 @@
-		function ($scope, $timeout, bookService, $state, $stateParams, $location) {
+	.controller('DataGridCtrl', ['$scope', '$timeout', 'bookService', '$state', '$stateParams', '$location', 'uiGridConstants', 'autoTags', 'authService',
+		function ($scope, $timeout, bookService, $state, $stateParams, $location, uiGridConstants, autoTags, authService) {
+			resizeGrid = function() {
+				var gridContainer = document.getElementsByClassName("gridStyle")[0];
+				var footer = document.getElementsByClassName("site-footer")[0];
 
-			$scope.filterOptions = {
-				filterText: "", // gets updated by user action in ngGrid
-				useExternalFilter: true
-			};
-			$scope.totalServerItems = 0;
-			$scope.pagingOptions = {
-				pageSizes: [20, 50, 100],
-				pageSize: 20, // Gets updated by user action in ngGrid
-				currentPage: 1 // Gets updated by user action in ngGrid
-			};
-			$scope.sortInfo = {
-				fields: [], directions: [], columns: []
+				gridContainer.style.height = (window.innerHeight - gridContainer.offsetTop - footer.offsetHeight) + "px";
 			};
-			$scope.getBookRange = function (count, currentPage, searchText) {
-				var first = (currentPage - 1) * count;
-				var sortField = $scope.sortInfo.fields[0];
-				var sortBy = null;
-				// Todo: setting sortBy to a complex field like this does not work...no sorting happens.
-				// It appears we will need to put a redundant top-level data field in the record for
-				// anything we want to sort by. This is especially annoying in that it may not actually
-				// be useful to sort by all the fields...but ng-grid allows the user to do so.
-				// Possibly we could disallow some fields with a message and switch to some other sort.
-				// Another reason a redundant field is probably necessary is that parse.com's built-in sorting
-				// is case-sensitive, and their only suggestion for overcoming this is a redundant
-				// all-lower-case field. Similarly any other sort-key field must somehow be in a form
-				// where a plain binary sort will give the right results.
-				if (sortField == 'title') {
-					sortBy = 'volumeInfo.title';
-				}
-				bookService.getFilteredBookRange(first, count, searchText, "", "", "", sortBy).then(function (result) {
-					$scope.visibleBooks = result;
-					$scope.visibleData = $scope.visibleBooks.map(function (item) {
-						return {
-							title: item.volumeInfo.title,
-							published: item.volumeInfo.publishedDate,
-							publisher: item.volumeInfo.publisher,
-							pages: item.volumeInfo.pageCount
-						};
+
+			window.addEventListener("resize", resizeGrid);
+
+			$scope.getBooks = function() {
+				var first = 0;
+				bookService.getFilteredBooksCount('', '', '', '', true).then(function (result) {
+					var count = result;
+
+					bookService.getFilteredBookRange(first, count, '', '', '', '', '', '', true).then(function (result) {
+						$scope.booksData = result.map(function (item) {
```

Comment:
I suspect you don't need two queries here. If you just pass a large number to getFilteredBookRange (I don't think parse.com will give you more than 1000 anyway), can you just get the count from result?